### PR TITLE
Fix bug in all_azure_dataset_ids and all_dataset_mountpoints

### PR DESCRIPTION
**Pull request by Shruthi42**

### CODE REVIEW

**Review by ant0nsc — APPROVED** [review]

File: InnerEye/ML/configs/segmentation/ProstatePaper.py [hunk 1]
```diff
@@ -11,6 +11,9 @@
 fg_classes = ["external", "femur_r", "femur_l", "rectum", "prostate", "bladder", "seminalvesicles"]
 fg_display_names = ["External", "Femur_R", "Femur_L", "Rectum", "Prostate", "Bladder", "SeminalVesicles"]
 
+# Change this string to the name of your dataset on Azure blob storage.
+AZURE_DATASET_ID = "your_dataset"
```
Comment (ant0nsc) [inline InnerEye/ML/configs/segmentation/ProstatePaper.py:15]: here and above - this somehow contravenes the suggestion of subclassing those models

File: InnerEye/ML/deep_learning_config.py [hunk 2]
```diff
@@ -309,18 +309,38 @@ class DatasetParams(param.Parameterized):
                        "AzureML. Use an empty string for all datasets where a randomly chosen mount/download point "
                        "should be used.")
 
+    def validate(self) -> None:
+        if not self.azure_dataset_id and self.local_dataset is None:
+            raise ValueError("Either of local_dataset or azure_dataset_id must be set.")
+
+        if not self.azure_dataset_id and len(self.extra_azure_dataset_ids) > 0:
+            raise ValueError("Please specify an azure dataset in azure_dataset_id "
+                             "before adding datasets to extra_azure_dataset_ids.")
```
Comment (ant0nsc) [inline InnerEye/ML/deep_learning_config.py:318]: maybe we are being too strict here? if someone specifies both datasets in extra..., and nothing in azure_dataset, it does not change the behaviour at all? The only thing that should matter: is it consistent across dataset_ids and mount points?

File: Tests/Common/test_segmentation_configs.py [hunk 3]
```diff
@@ -212,7 +186,8 @@ def test_head_and_neck_paper_with_some_ground_truth_ids(
     Check that passing a num_structures between 1 and len(defaults) generates the correct subset.
     """
     ground_truth_ids = DEFAULT_HEAD_AND_NECK_GROUND_TRUTH_IDS[:ground_truth_count]
-    config = HeadAndNeckPaper(num_structures=ground_truth_count)
+    config = HeadAndNeckPaper(local_dataset=Path("foo"),
```
Comment (ant0nsc) [inline Tests/Common/test_segmentation_configs.py:189]: Do you still need that extra constructor arg "local_dataset=...", now that you hardcoded the azure dataset ID in the model? (see my comment above - I think actually this change here is better than providing a fake Azure dataset ID)

**Review by ant0nsc — APPROVED** [review]

File: InnerEye/ML/configs/segmentation/ProstatePaper.py (hunk 1 above)
Comment (Shruthi42) [reply InnerEye/ML/configs/segmentation/ProstatePaper.py]: Removed `AZURE_DATASET_ID` from these classes.

**Review by melanibe — APPROVED** [review]

**Review by ant0nsc — APPROVED** [review]

**Pull request merged**